### ISSUE TITLE
можно ли изменить алгоритм для модификации всех пакетов?

### ISSUE THREAD

**Issue by koroni** [Waujito/youtubeUnblock#129]
Здравствуйте

Обьясню задачу. 

Наш провайдер запрещает раздавать интернет, раньше лечилось изменением ttl теперь они включили DPI и каким то образом блокируют интернет после анализа. Требуется модифицировать все исходящие пакеты WAN для усложенения DPI.

В вашем модуле модифицируются только HELLO (если я правильно понял). 

Т.к. я глубоко в код не закапывался, на сколько будет сложно добавить например параметр full_mode=1 после включения которого, абс все пакеты будут изменяться?

**Comment by Waujito** [comment]
А точно ли DPI? Поведение телефона от пк мало чем отличается в плане интернета. Там из адекватных остаётся разве что ttl и http user agent. В первом случае мы можем просто менять ttl, в во втором заблокировать http в браузере. А анблок с sni domains all в целом может скрыть от вашего провайдера домены

**Comment by koroni** [comment]
точно знаю что когда включаешь винду через роутер, сразу блокировка наступает (падает скорость до нуля)
всталяешь в телефон - все работает!

с ttl все далал: жестко прибивал 128 в роутере

и в догонку, у меня не блокировка срабатывала, а падение скорости "склизкий момент такой )))"

**Comment by Waujito** [comment]
А что если нагло заблокировать 80 порт и поставить анблок во все домены? Тогда вряд ли они смогут что-либо распознать. Еще можно попробовать поставить doh на роутер, мало ли. Кстати, а на телефоне скорость падает при раздаче?

**Comment by koroni** [comment]
да, точно забыл рассказать, если с телефона поставить раздачу, что провайтер фиксирует раздачу и прикладывает "доп. оплату к тарифу" ))) - "за раздачу"

**Comment by Waujito** [comment]
> да, точно забыл рассказать, если с телефона поставить раздачу, что провайтер фиксирует раздачу и прикладывает "доп. оплату к тарифу" ))) - "за раздачу". В телефоне я не фиксировал ttl

Показать бы ФАС этого провайдера... 
О ком речь кстати?

**Comment by koroni** [comment]
это один из операторов сотовой связи, но дабы не обретать публичность, к сожалению могу рассказать, могу только в личку.

**Comment by Waujito** [comment]
Да не, я хотел узнать Билайн это или не Билайн. Они точно таким баловались. Еще вопрос как меняете ttl? Если правилами, скиньте nft list ruleset с роутера

**Comment by koroni** [comment]
ttl менял прошивкой модема (глубже не закапывался)

**Comment by Waujito** [comment]
Сомнительная вещь, если честно. Подключите по usb к роутеру телефон, раздайте с него интернет по usb и дайте роутеру правила
`nft add rule inet fw4 mangle_postrouting ip ttl set 65`
`nft add rule inet fw4 mangle_postrouting ip6 hoplimit set 65`

**Comment by koroni** [comment]
там внутри модема скорее всего так и сделано. 
НО моя цель масировать пакеты, что бы провайду не получалось прервать SSL рукопожатие. 
поведение следующее:
1. если https то рубится SSL  - ECONNRESET
2. если по http заходить то происходит безусловный редирект на страницу с текстом "раздавать запрещено"
3. пробовал менять user-agent: не помогает

у вас есть по рукой готовый инструмент анализа (подгонки запроса) что бы пройти защиту?

**Comment by Waujito** [comment]
Если в одно время с работой винды ткнуться с другого устройства, он пустит?

И заблокируйте всё-таки для теста хттп 80 + анблок на all domains.
